grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java

```diff
@@ -252,10 +257,19 @@ protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req) throws
         ctx.logBuilder().defer(RequestLogProperty.REQUEST_CONTENT,
                                RequestLogProperty.RESPONSE_CONTENT);
 
-        final HttpResponseWriter res = HttpResponse.streaming();
-        final ArmeriaServerCall<?, ?> call = startCall(
-                registry.simpleMethodName(method.getMethodDescriptor()), method, ctx, req, res,
-                serializationFormat);
+        final ServerCall<?, ?> call;
+        final HttpResponse res;
+        if (method.getMethodDescriptor().getType() == MethodType.UNARY) {
```

Comment:
Ignore the comment avoid. 
As `UnaryServerCall` provides an optimized implementation for unary calls, `CLIENT_STREAMING` is still handled by `StreamingServerCall`.
This condition is added to inject an `HttpResponse` optimized for a non-streaming response.
We may provide a lighter implementation rather than a combination of a fixed `HttpResponse` and deferred `HttpResponse`.